File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/security/AuthAwareAuthenticationToken.java
package uk.gov.justice.digital.hmpps.keyworker.security;

import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.oauth2.jwt.Jwt;
import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;

import java.util.Collection;

@Getter
public class AuthAwareAuthenticationToken extends JwtAuthenticationToken {
    private final Object principal;
    private final String name;

    AuthAwareAuthenticationToken(final Jwt jwt, final Object principal, final Collection<? extends GrantedAuthority> authorities) {
        super(jwt, authorities);
        this.principal = principal;
        this.name = principal != null ? principal.toString() : "";
    }
}

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/security/AuthAwareAuthenticationToken.kt
package uk.gov.justice.digital.hmpps.keyworker.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.oauth2.jwt.Jwt
import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken

internal class AuthAwareAuthenticationToken(jwt: Jwt, private val principal: Any?, authorities: Collection<GrantedAuthority>) : JwtAuthenticationToken(jwt, authorities) {

    private val name: String =  principal?.toString() ?: ""

    override fun getPrincipal(): Any? {
        return this.principal
    }

    override fun getName(): String {
        return this.name
    }

}